tests/tasks/labware_location/test_main.py

```diff
@@ -0,0 +1,451 @@
+
+    bad_entry = TEST_ENTRIES["bad_input_entry_without_location"]["input_lw"]["labwares"][0]
+
+    assert mock_error.has_calls(call(f"Found invalid entry: {bad_entry}"))
+
+
+@pytest.mark.parametrize(
+    "lw_labwares_table",
+    [TEST_ENTRIES["bad_input_entry_without_audits"]["input_lw"]["labwares"]],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "lw_locations_table",
+    [TEST_ENTRIES["bad_input_entry_without_audits"]["input_lw"]["locations"]],
+    indirect=True,
+)
+@patch("logging.info")
+@patch("logging.error")
+def test_given_bad_input_entry_without_audits_when_sorting_entries_then_check_entry_filtered_out(
+    mock_info,
+    mock_error,
+    config,
+    mlwh_labware_locations_table,
+    lw_labwares_table,
+    lw_audits_table,
+    lw_users_table,
+    lw_locations_table,
+    lw_coordinates_table,
+):
+    sync_changes_from_labwhere(config)
+    assert mock_info.has_calls(
+        call("Starting sync labware locations task..."),
+        call("Updating 0 rows..."),
+        call("Closing connections to databases..."),
+        call("Task successful!"),
+    )
+
+    bad_entry = TEST_ENTRIES["bad_input_entry_without_audits"]["input_lw"]["labwares"][0]
+
+    assert mock_error.has_calls(call(f"Found invalid entry: {bad_entry}"))
+
+
+@pytest.mark.parametrize(
+    "lw_labwares_table",
+    [TEST_ENTRIES["bad_input_entry_without_location_without_audits"]["input_lw"]["labwares"]],
+    indirect=True,
+)
+@patch("logging.info")
+@patch("logging.error")
+def test_given_bad_input_entry_without_location_without_audits_when_sorting_entries_then_check_entry_filtered_out(
+    mock_info,
+    mock_error,
+    config,
+    mlwh_labware_locations_table,
+    lw_labwares_table,
+    lw_audits_table,
+    lw_users_table,
+    lw_locations_table,
+    lw_coordinates_table,
+):
+    sync_changes_from_labwhere(config)
+    assert mock_info.has_calls(
+        call("Starting sync labware locations task..."),
+        call("Updating 0 rows..."),
+        call("Closing connections to databases..."),
+        call("Task successful!"),
+    )
+
+    bad_entry = TEST_ENTRIES["bad_input_entry_without_location_without_audits"]["input_lw"]["labwares"][0]
+
+    assert mock_error.has_calls(call(f"Found invalid entry: {bad_entry}"))
+
+
+@pytest.mark.parametrize(
+    "lw_labwares_table",
+    [TEST_ENTRIES["mixed_input_entries"]["input_lw"]["labwares"]],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "lw_audits_table",
+    [TEST_ENTRIES["mixed_input_entries"]["input_lw"]["audits"]],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "lw_users_table",
+    [TEST_ENTRIES["mixed_input_entries"]["input_lw"]["users"]],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "lw_locations_table",
+    [TEST_ENTRIES["mixed_input_entries"]["input_lw"]["locations"]],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "lw_coordinates_table",
+    [TEST_ENTRIES["mixed_input_entries"]["input_lw"]["coordinates"]],
+    indirect=True,
+)
```

Comment:
To avoid confusion, I have created a single fixture to clear all the tables before each test. I have also added a helper functions which writes to all the relevant tables. Hopefully it is easier to understand what each test is doing now 😄